CounterStrike/src/dev/danablend/counterstrike/listeners/EntityPickupItemListener.java

```diff
@@ -45,33 +44,41 @@ public void onPlayerPickupItem(EntityPickupItemEvent e) {
             }
 
             if (cp.getTeam().equals(TeamEnum.TERRORISTS) && item.getType().equals(Material.TNT)) {
-                e.getItem().remove();
-                p.getInventory().setItem(4, CSUtil.getBombItem());
+                //can continue
                 return;
+            } else if (cp.getTeam().equals(TeamEnum.COUNTER_TERRORISTS) && item.getType().equals(Material.TNT)) {
+                e.setCancelled(true);
             }
 
             if (Weapon.isWeapon(item)) {
                 Weapon weapon = Weapon.getByItem(item);
                 switch (weapon.getWeaponType()) {
                     case RIFLE:
                         if (p.getInventory().getItem(0) == null) {
-                            p.getInventory().setItem(0, item);
                             e.getItem().remove();
+                            p.getInventory().setItem(0, item);
                         }
                         e.setCancelled(true);
-                        break;
+

```

Comment:
Took some time to remember, the problem of not documenting when we find a problem...
This is because you always want to allow to pickup a grenade, but you don't want to for a rifle or pistol if you already have one

Reply:
I think you didn't get my point. If you have such switch:
```switch(status) {
case ALIVE {
 System.out.println("I'm alive!");
}

case DEAD {
 System.out.println("I'm dead!");
}
}```

When status is DEAD, it will print
`I'm dead!`

BUT if status will be ALIVE, it will print both messages:
`I'm alive!
I'm dead!`.

We need `break` after each case, because instead every case, that is below is called. (see for example https://www.reddit.com/r/learnjava/comments/susdi2/why_does_switch_without_break_execute_all_cases/).